### PR TITLE
Check if setting is disabled on default scope

### PR DESCRIPTION
**Pull request by kassner**
Work in progress

### Description

This PR fixes 22890, but it will still have the same issue when the value is defined on _website_ level only. I'm taking suggestions on how to fix this.

Given the way paths are designed, it's not possible to guess the website code from the store code, so I believe eventually I'll have to load `Magento\Store\Model\Store` somewhere to get that information.

### Fixed Issues

1. #22890: Disabled config can be overwritten via admin

### Manual testing scenarios

See issue description.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by kassner** [comment]
If anyone wants to give it a shot, add this item to the `isReadonlyDataProvider`:

```php
[
    'path' => 'general/web/locale',
    'scope' => 'store',
    'scopeCode' => 'myStore',
    'confValue' => null,
    'variables' => [],
    'configMap' => [
        [
            'system/default/general/web/locale',
            null,
            null,
        ],
        [
            'system/website/myWebsite/general/web/locale',
            null,
            'website_value',
        ],
    ],
    'expectedResult' => true,
],
```

**Comment by phoenix128** [comment]
Thank you @kassner ,
I just tested it locally and it seems to be fixing the issue.

Thank you very much for your contribution.
Just waiting for static testing to run to approve your PR.

**Comment by phoenix128** [comment]
Hello @kassner ,
you have a single static test failing. It is not caused by your PR, but it involves a couple of files you changed.
Can you please FIX them in the scope of this pull request?

```
PHP Code Sniffer detected 2 violation(s): 

FILE: ...to/Config/Model/Config/Reader/Source/Deployed/SettingChecker.php
----------------------------------------------------------------------
FOUND 4 ERRORS AFFECTING 3 LINES
----------------------------------------------------------------------
  92 | ERROR | [x] Extra @param found in method annotation
  92 | ERROR | [x] $scopeCode duplicate found in method annotation
 106 | ERROR | [ ] Direct use of $_ENV Superglobal detected.
 107 | ERROR | [ ] Direct use of $_ENV Superglobal detected.
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------


FILE: .../Unit/Model/Config/Reader/Source/Deployed/SettingCheckerTest.php
----------------------------------------------------------------------
FOUND 4 ERRORS AND 1 WARNING AFFECTING 3 LINES
----------------------------------------------------------------------
  78 | WARNING | [ ] Line exceeds maximum limit of 120 characters;
     |         |     contains 126 characters
  99 | ERROR   | [x] Opening parenthesis of a multi-line function
     |         |     call must be the last content on the line
  99 | ERROR   | [x] Opening parenthesis of a multi-line function
     |         |     call must be the last content on the line
 105 | ERROR   | [x] Closing parenthesis of a multi-line function
     |         |     call must be on a line by itself
 105 | ERROR   | [x] Closing parenthesis of a multi-line function
     |         |     call must be on a line by itself
----------------------------------------------------------------------
PHPCBF CAN FIX THE 4 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------



Failed asserting that 2 matches expected 0.
```

Thank you

File: app/code/Magento/Config/Test/Unit/Model/Config/Reader/Source/Deployed/SettingCheckerTest.php [hunk 1]
```diff
@@ -95,13 +96,13 @@ public function testIsReadonly($path, $scope, $scopeCode, $confValue, array $var
 
         $this->configMock->expects($this->any())
             ->method('get')
-            ->willReturnMap([
+            ->willReturnMap(array_merge([
```
Comment (phoenix128) [inline app/code/Magento/Config/Test/Unit/Model/Config/Reader/Source/Deployed/SettingCheckerTest.php]: Please fix the codestyle accordingly to the following message:

```
FILE: .../Unit/Model/Config/Reader/Source/Deployed/SettingCheckerTest.php
----------------------------------------------------------------------
FOUND 4 ERRORS AND 1 WARNING AFFECTING 3 LINES
----------------------------------------------------------------------
  78 | WARNING | [ ] Line exceeds maximum limit of 120 characters;
     |         |     contains 126 characters
  99 | ERROR   | [x] Opening parenthesis of a multi-line function
     |         |     call must be the last content on the line
  99 | ERROR   | [x] Opening parenthesis of a multi-line function
     |         |     call must be the last content on the line
 105 | ERROR   | [x] Closing parenthesis of a multi-line function
     |         |     call must be on a line by itself
 105 | ERROR   | [x] Closing parenthesis of a multi-line function
     |         |     call must be on a line by itself
----------------------------------------------------------------------
```

**Review by phoenix128 — CHANGES_REQUESTED** [review]
Thank you for your contribution, would you please check my comments?

**Comment by kassner** [comment]
@phoenix128 I ran phpcbf on it, I hope it's enough.

Bear in mind that this PR does not fix the full extent of the issue, as I stated in the PR's description.

**Comment by phoenix128** [comment]
Hello @kassner ,
seems like phpcbf did not fix all the issues.

File: app/code/Magento/Config/Model/Config/Reader/Source/Deployed/SettingChecker.php [hunk 2]
```diff
@@ -64,6 +64,12 @@ public function isReadOnly($path, $scope, $scopeCode = null)
             $this->placeholder->generate($path, $scope, $scopeCode)
         );
 
+        if (null === $config) {
```
Comment (phoenix128) [inline app/code/Magento/Config/Model/Config/Reader/Source/Deployed/SettingChecker.php]: Shouldn't this be after the specific scope check (i mean after line 75)?

**Review by phoenix128 — CHANGES_REQUESTED** [review]

**Review by sidolov — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sidolov, thank you for the review.
[ENGCOM-5441](https://jira.corp.magento.com/browse/ENGCOM-5441) has been created to process this Pull Request
:eight_spoked_asterisk: @sidolov, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

File: app/code/Magento/Config/Model/Config/Reader/Source/Deployed/SettingChecker.php (hunk 2 above)
Comment (engcom-Foxtrot) [reply app/code/Magento/Config/Model/Config/Reader/Source/Deployed/SettingChecker.php]: @phoenix128 fixed.

**Review by phoenix128 — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @phoenix128, thank you for the review.
[ENGCOM-5441](https://jira.corp.magento.com/browse/ENGCOM-5441) has been created to process this Pull Request

**Pull request merged**